### ISSUE TITLE
(🐞) False error with assignment of generic union of types to the same union of types

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#15733]
```py
from typing import TypeVar

class A: pass
class B: pass

T_AB = TypeVar("T_AB", bound=A | B)

def foo(t: type[T_AB]) -> None:
    x: type[A | B] = t  # error
```

It works when the `type` is in the bound:
```py
from typing import TypeVar

class A: pass
class B: pass

T_AB = TypeVar("T_AB", bound=type[A | B])

def foo(t: T_AB) -> None:
    x: type[A | B] = t  # no error
```
